parlai/core/params.py

```diff
@@ -1043,10 +1055,12 @@ def _kwargs_to_str_args(self, **kwargs):
                 string_args.append(last_option_string)
             elif isinstance(action, argparse._StoreAction) and action.nargs is None:
                 string_args.append(last_option_string)
-                string_args.append(str(value))
+                string_args.append(self._value2argstr(value))
             elif isinstance(action, argparse._StoreAction) and action.nargs in '*+':
                 string_args.append(last_option_string)
-                string_args.extend([str(v) for v in value])
+                string_args.extend(
+                    [self._value2argstr(value, space=True) for v in value]
```

Comment:
@stephenroller space is still in use here. I am not confident that it is actually being used anywhere though, would you know if this case is ever hit ?
If not I can get rid of it

Reply:
The two sections should mirror each other